Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1247,6 +1247,112 @@ The following approach can be used in order to patch the JavaScript file:
 
 #### Dynamic Instrumentation
 
+##### Information Gathering
+
+In this section we will learn about how to use Frida to obtain information about a running application.
+
+###### Getting Loaded Classes and their Methods
+
+In the Frida REPL to access Java runtime, `Java` command can be used to access information within the running app. Unlike `ObjC` command for iOS, for Java, the code need to be called via `Java.perform()` function. Thus, the best way to get list of loaded Java classes and their corresponding methods and field is to use Frida scripts. One such script is listed below.
```

Comment:
I've kept Sven's suggestions plus some enhancements:

- Frida CLI instead of REPL ( we've introduced it with that name as in the Frida docs https://www.frida.re/docs/frida-cli/)
- remove the ( ) in the method
- some rephrasing 

```suggestion
You can use the command `Java` in the Frida CLI to access the Java runtime and retrieve information from the running app. Remember that, unlike Frida for iOS, in Android you need to wrap your code inside a `Java.perform` function. Thus, it's more convenient to use Frida scripts to e.g. get a list of loaded Java classes and their corresponding methods and fields or for more complex information gathering or instrumentation. One such scripts is listed below.
```